InterfaceMATLAB/tsgDifferentiate.m

```diff
@@ -0,0 +1,56 @@
+function [result] = tsgDifferentiate(lGrid, mX)
+%
+% [result] = tsgEvaluate(lGrid, points)
+%
+% differentiates the interpolant at the points of interest and returns the result
+% this should be called after the grid has been created and after values
+% have been loaded
+%
+% INPUT:
+%
+% lGrid: a grid list created by tsgMakeXXX(...)
+%
+% mX: an array of size [num_x, dimensions]
+%     specifies the points where the interpolant should be differentiated
+%     Note: do not confuse points here with the nodes of the grid
+%               here points are user specified points to differentiate the
+%               interpolant (or approximation)
+%
+% OUTPUT:
+%
+% result: an array of size [num_x, iOut * dimensions]
```

Comment:
Ugh, this is now an ugly problem thanks to MATLAB.
* the output of differentiate is not a derivative or a gradient, but a Jacobian (multi-input/multi-output case)
* in C and C++, we use 1D arrays for matrices anyway, so we this is OK
* MATLAB and Python have native multi-d arrays, so we should respect that
* in Python, we do one derivative at a time, so we just return a matrix (right? we should)
* in MATLAB, there are no 3D arrays, but there are lists of matrixes
```
A = [1, 2; 3, 4]; B = [5, 6; 7, 8];
C = {A, B};
C[1]
C[2]
```
* basically, if `num_x == 1`, we should just return the 2D Jacobian, otherwise we should list of matrices as above.

Reply:
Done.